sdk/core/core-http/lib/policies/deserializationPolicy.ts

```diff
@@ -140,140 +140,131 @@ export function deserializeResponseBody(
   response: HttpOperationResponse
 ): Promise<HttpOperationResponse> {
   return parse(jsonContentTypes, xmlContentTypes, response).then((parsedResponse) => {
-    const shouldDeserialize: boolean = shouldDeserializeResponse(parsedResponse);
-    if (shouldDeserialize) {
-      const operationSpec: OperationSpec | undefined = parsedResponse.request.operationSpec;
-      if (operationSpec && operationSpec.responses) {
-        const statusCode: number = parsedResponse.status;
-
-        const expectedStatusCodes: string[] = Object.keys(operationSpec.responses);
-
-        const hasNoExpectedStatusCodes: boolean =
-          expectedStatusCodes.length === 0 ||
-          (expectedStatusCodes.length === 1 && expectedStatusCodes[0] === "default");
+    const shouldDeserialize = shouldDeserializeResponse(parsedResponse);
+    if (!shouldDeserialize) {
```

Comment:
>shouldDeserialize [](start = 9, length = 17)

nit: maybe remove this local variable as the method name indicates clearly what would be the result

Reply:
Fixed